cmd/credentialUtil.go

```diff
@@ -298,6 +298,20 @@ func checkAuthSafeForTarget(ct common.CredentialType, resource, extraSuffixesAAD
 		return nil
 	case common.ECredentialType.OAuthToken(),
 		common.ECredentialType.SharedKey():
+		// Files doesn't currently support OAuth.
+		// TODO: When adding files OAuth support to AzCopy eventually, fix this if statement.
+		if resourceType != common.ELocation.Blob() && resourceType != common.ELocation.BlobFS() {
+			// There may be a reason for files->blob to specify this.
+			// Files doesn't take OAuth currently, anyway.
```

Comment:
Why do we need to consider Azure Files as a special case here?  Why can't we just do if resourceType == common.ELocation.Local() at the very first if statement (the one that currently looks at !Blob && !BlobFS) and then just return nil.  (And that would be the whole thing).

I.e. I'm asking why we can't let the Files locations just run through the host name test. (All Files urls will pass those checks, since the hard-coded domain suffixes deliberately dont' restrict to blob or blobFS).

That way, there would be nothing here that we would need to remember to maintain in future.  Comments of the nature "TODO: when we do X later (in some other part of the codebase) remember to change this" tend to be dangeorous when it comes to security, since they can be forgotten.  (They're equally likely to be overlooked in non-security situations - its just that the consequences are more severe with security).

Reply:
Good point that it does match the pattern. No reason to have a special case here.